rfcs/text/0000-multi-data-collection-dataset-processing.md

```diff
@@ -0,0 +1,233 @@
+5. Decouples analysis initiation from data representation allowing for future flexibility
+
+## General Metadata Requirements
+In the above framework, analysis pipelines will receive a set of bundles that _potentially_ need to be co-processed. As aforementioned, this signal will not explicitly identify the bundles that need to be co-processed in a single pipeline, instead, it will serve as a notification **indicating that any co-processing is to be done only within the boundaries of this set of files (submission envelope). **
+
+Analysis can initiate multiple pipeline invocations as a result of a single DAPS being submitted.
+
+The actual grouping of the files is to be performed by analysis via a search in the metadata of the bundles submitted following well-defined modality-specific set of rules. For this reason, the metadata must satisfy the following metadata requirements (MR):
+
+Furthermore, if part of the data submitted in a DAPS has already been processed the Analysis infrastructure will only re-process the parts that have not been modified.
+
+1. The data modality of each data file included in the DAPS must be clearly identifiable (MR1)
+2. Data that need to be processed together must be clearly identifiable by a modality-specific set of metadata variables that will group the files according to processing requirements. (MR2)
+3. Among the data that needs to be processed together, the metadata must provide sufficient information to correctly order, pair or otherwise establish correspondence and type of the input files. (MR3)
+
+## Proposed Signalling Implementation
+The DAPS proposal above only sets specific requirements for the signal to initiate data analysis and minimum requirements for the associated metadata. We hereby provide a specific recommendation for implementation via bundle notifications.
+
+We recommend that the data store (DSS) is used to store DAPS as a new bundle type of **“DAPS Manifest”**, and that the current search-based notification system is adjusted so that instead of listening for new primary bundles, analysis listens for new DAPS bundles that are exclusively used for the initiation of pipelines. This implementation has two major advantages: (1) the DAPS are permanently saved at the main storage site for the HCA project (DSS) so the analyses can be reproducibly triggered and (2) The existing notification mechanisms are reused, avoiding the need for implementation of a new out-of-band notification system.
+
+We propose that as an initial implementation approach DAPS are used at set data aggregation levels (e.g. sample, project) but critically the implementation must be flexible to arbitrary data aggregation levels to accommodate future project needs. We propose that triggering is initially done manually by data wranglers and/or ingest, but we envisage transition to automatic triggering in the future.
+
+## Example: Analysis Implementation for a DAPS
```

Comment:
I think it would be helpful if this example had more detail. Most of this RFC is pretty abstract, deliberately eliding implementation details. But, it's hard to understand what exactly you have in mind without a worked example. For instance, in this 10x case who would create the DMB and how would they do that? What if additional sequencing was done later? Where does DM parsing code live, and who writes/maintains it? What's the lifecycle of a DMB?